pkg/comp-functions/runtime/function_mgr.go

```diff
@@ -1127,7 +1142,60 @@ func (s *ServiceRuntime) UsageOfBy(of, by string) error {
 		},
 	}
 
-	return s.SetDesiredKubeObject(usage, name)
+	composedRes, err := composed.From(usage)
+	if err != nil {
+		return fmt.Errorf("cannot convert usage object to managed resource: %w", err)
+	}
+
+	s.desiredResources[resource.Name(name)] = &resource.DesiredComposed{Resource: composedRes}
+
+	return nil
+}
+
+// unwrapUsage will check if the usage is wrapped into an object.
+// It will set the wrapping obect to only observe it. Then on the next reconcile
+// after it ensures that the right management policy is set, it will not add it
+// to the desired state again. And finally on the third reconcile it will
+// return true to indicate that it got removed properly. Allowing Crossplane
+// to adopt the Usage objects without wrapping in a kube object.
+func (s *ServiceRuntime) unwrapUsage(name string) (bool, error) {
+	usage := &xkube.Object{}
+	err := s.GetObservedComposedResource(usage, name)
+	if err != nil {
+		if err == ErrNotFound {
+			return true, nil
+		}
+		return false, err
+	}
+
+	resources, err := request.GetObservedComposedResources(s.req)
+	if err != nil {
+		return false, err
+	}
+	res := resources[resource.Name(name)]
+	if res.Resource.GetKind() != "Object" {
+		return true, nil
+	}
+
+	// we need to clean the objectmeta, or Crossplane will struggle with adding
+	// it to the `resourceRefs` array.
+	usage.ObjectMeta = metav1.ObjectMeta{
+		Name:            usage.GetName(),
+		Annotations:     usage.Annotations,
+		Labels:          usage.Labels,
+		OwnerReferences: usage.OwnerReferences,
+	}
+
+	if len(usage.Spec.ManagementPolicies) == 0 || usage.Spec.ManagementPolicies[0] != xpv1.ManagementActionObserve {
```

Comment:
also here, if managementpolicies for some reason will be nil, then You will receive:
```
panic: runtime error: index out of range [0] with length 0
```